### ISSUE TITLE
About loss_c in Multibox Loss

### ISSUE THREAD

**Issue by taneslle** [lzx1413/PytorchSSD#14]
Hi~
[https://github.com/lzx1413/PytorchSSD/blob/cd5776d7dd348f9c6a996cec8accf749e0bffee0/layers/modules/multibox_loss.py#L93](url)
This operation seems to me that same as to calculate the softmax cross entropy loss, so why not use torch.nn.functional.cross_entropy after softmax directly?
ps. `+x_max` and `-x_max` in log_sum_exp seems do nothing either.

**Comment by USTClj** [comment]
@taneslle , HI, have you figured it out why?

**Comment by lzx1413** [comment]
Because we want to get the softmax loss for each element. However, in the early version of Pytorch, this function is not supported and only the sum or average loss of all the elements is produced.

**Comment by lzx1413** [comment]
You can calculate the softmax loss once and generate a mask for postive instances and hard negtive instances. Then you can multiply them together to generate the valuable loss.

**Comment by USTClj** [comment]
Yes. Another issue is that if we can do it by using the F.cross_entropy twice. Will it change the input tensor's gradient because the forward pass of the same input is tracked. Thank you for your help again!

**Comment by lzx1413** [comment]
If you don't add the first softmax's loss to the final total loss, it will not effect the network.

**Comment by USTClj** [comment]
Thank you.
